### DISCUSSION TITLE
Operation Name is passed as empty string in DefaultDgsWebfluxHttpHandler

### DISCUSSION THREAD

**Discussion by PrasoonJain** [Netflix/dgs-framework#884] [Q&A]
Hey There,

I am trying to use operation_name field to identify latencies of different queries through metrics.

But it is hardcoded as empty string in "DefaultDgsWebfluxHttpHandler" file line-56 while calling execute

    override fun graphql(request: ServerRequest): Mono<ServerResponse> {
        @Suppress("UNCHECKED_CAST") val executionResult: Mono<ExecutionResult> =

            request.bodyToMono(String::class.java)
                .flatMap { body ->
                    if ("application/graphql" == request.headers().firstHeader("Content-Type")) {
                        Mono.just(QueryInput(body))
                    } else {
                        Mono.fromCallable {
                            val readValue = mapper.readValue<Map<String, Any>>(body)
                            val query: String? = when (val iq = readValue["query"]) {
                                is String -> iq
                                else -> null
                            }
                            QueryInput(
                                query,
                                (readValue["variables"] ?: emptyMap<String, Any>()) as Map<String, Any>,
                                (readValue["extensions"] ?: emptyMap<String, Any>()) as Map<String, Any>,
                            )
                        }
                    }
                }
                .flatMap { queryInput ->
                    logger.debug("Parsed variables: {}", queryInput.queryVariables)
                    dgsQueryExecutor.execute(
                        queryInput.query,
                        queryInput.queryVariables,
                        queryInput.extensions,
                        request.headers().asHttpHeaders(),
                        "",
                        request
                    )
                }`

Any specific reason on why is it so ?

**Comment by ptorkko** [comment]
was also wondering about this

there's an open bug for it https://github.com/Netflix/dgs-framework/issues/672

**Reply by srinivasankavitha** [reply]
Thanks. Yes, we have an open issue. However, we have not been able to address it yet. Feel free to open a PR if this is a blocking issue for you, we do welcome contributions.

**Reply by PrasoonJain** [reply]
https://github.com/Netflix/dgs-framework/pull/941 - Have raised a fix for this.
@srinivasankavitha Can you approve build check runs and help review ?

**Reply by PrasoonJain** [reply]
https://github.com/Netflix/dgs-framework/pull/941 - Hey @srinivasankavitha , I have re-pushed a commit with build checks installed and fix. Can you please approve CI checks again. Apologies for to and fro.
